### PR TITLE
Mixing with ComfortableInventory

### PR DESCRIPTION
**Pull request by kohanis**

### CODE REVIEW

**Comment by kohanis** [comment]
Any questions, suggestions?

**Comment by thmsndk** [comment]
I'm swarmed at work right now, I will try and look trough it ASAP, probably in the weekend :)

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(CommonPostfix));
+
+        private static readonly HarmonyMethod inventory_RemoveItem_Prefix =
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(Inventory_RemoveItem_Prefix));
+
+        private static readonly HarmonyMethod blockCreator_CreateBlock_Prefix =
+            new HarmonyMethod(typeof(Slot_RemoveItem_MultiPatch), nameof(BlockCreator_CreateBlock_Prefix));
+
+        internal static void Patch(Harmony harmony)
+        {
```
Comment (thmsndk) [inline InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs:24]: I'd love some more comments on the intentions of theese patches.
I can see you have patched a lot of lower level methods, I assume it's to catch a lot of different item types at a lower level, so new items are automatically supported as long as they follow the same rules?

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            harmony.Patch(PlayerInventory_RemoveSelectedHotSlotItem_Target, commonPrefix, commonPostfix);
```
Comment (thmsndk) [inline InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs]: I don't have the raft code on my backbone, but I assume this is patched for when for example your hook is breaking and it needs to be replaced with a new one?

File: InventoryTweaks/Patches/Helpers.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+                var slotItemInstance = localSlot.itemInstance;
+                if (slotItemInstance.UniqueIndex != originalUniqueIndex)
+                    continue;
+
+                if (slot.IsEmpty)
+                    PlayerInventory_MoveSlotToEmpty(inventory,
+                        new object[] { localSlot, slot, slotItemInstance.Amount });
+                else
+                    PlayerInventory_SwitchSlots(inventory, new object[] { localSlot, slot });
```
Comment (thmsndk) [inline InventoryTweaks/Patches/Helpers.cs]: Does this still remove the old item, if we swap the "broken" item with a new one from the inventory?

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            harmony.Patch(Inventory_RemoveItem_Target, inventory_RemoveItem_Prefix, commonPostfix);
```
Comment (thmsndk) [inline InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs]: Should it not be enough to patch PlayerInventory_RemoveSelectedHotSlotItem_Target? What is the reasoning for patching Inventory.RemoveItem?

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            harmony.Patch(BlockCreator_CreateBlock_Target, blockCreator_CreateBlock_Prefix, commonPostfix);
```
Comment (thmsndk) [inline InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs]: I am a little unsure what patching CreateBlock does, is it to refill for example purifiers when wanting to place a lot of them? I would think some of the other patches would catch thoose as well though 🤔

**Review by thmsndk — COMMENTED** [review]
All in all, looks like a solid PR, I'd like some clarifications on parts of it, it's done a different way than how I would do it, but I am by no means a harmony wizard :)

You can edit the modinfo.json to include you as an author as well :) and I'll make sure to edit the mod page to give you credits for the features :)

**Comment by kohanis** [comment]
Changed to transpilers in the end, but it should be way faster

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs (hunk 5 above)
Comment (kohanis) [reply InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs]: Yes, it removes item from hotslot when placing items such as purifiers or nets.

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs (hunk 4 above)
Comment (kohanis) [reply InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs]: Well, this is when consumable item is first found in hotslot. And, actually, I was thinking about option of replenishing any hotbar items, not just selected one

File: InventoryTweaks/Patches/Helpers.cs (hunk 3 above)
Comment (kohanis) [reply InventoryTweaks/Patches/Helpers.cs]: It's only for items with empty state, like water bottle

File: InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs (hunk 2 above)
Comment (kohanis) [reply InventoryTweaks/Patches/Slot_RemoveItem_MultiPatch.cs]: Added comment in new edition
Battery insertion, tank/purifier collection with empty water holder and shark bait throwing
```
Battery.Insert
Block_CookingStand_Purifier.CollectItem
SharkBaitThrower.OnHitWater
Tank.CollectItem
```

**Comment by kohanis** [comment]
Final refactoring and a few new features.
[Hold-Drop implementation](https://github.com/kohanis/raft-comfortable-inventory/commit/1c299df064175491d88dca14ce8fd15d0c2865df)
[PlayerInventory.ShiftMoveItem from/to hotbar fix](https://github.com/kohanis/raft-comfortable-inventory/commit/989528e2b695e41bae506ad02625d7cb52024b72)
[Ctrl Shift Move reimplementation](https://github.com/kohanis/raft-comfortable-inventory/commit/a737884948ac49a7422ec9f0cec0664be25a33e8)
Not sure about last, but my version works from/to hotbar

**Comment by thmsndk** [comment]
I'll see if I can find some time to look over it this weekend, you've reworked things quite a bit since I last reviewed it :D

**Comment by thmsndk** [comment]
Hmm, perhaps we should just include the tooltip on hover feature from this mod in your mod, and deprecate inventory tweaks? you seem to have more time to work on it than me?

**Pull request closed without merging**